libs/deepagents-cli/deepagents_cli/image_utils.py

```diff
@@ -0,0 +1,209 @@
+"""Utilities for handling image paste from clipboard."""
+
+import base64
+import io
+import os
+import subprocess
+import sys
+import tempfile
+from dataclasses import dataclass
+
+from PIL import Image
+
+
+@dataclass
+class ImageData:
+    """Represents a pasted image with its base64 encoding."""
+
+    base64_data: str
+    format: str  # "png", "jpeg", etc.
+    placeholder: str  # Display text like "[image 1]"
+
+    def to_message_content(self) -> dict:
+        """Convert to LangChain message content format.
+
+        Returns:
+            Dict with type and image_url for multimodal messages
+        """
+        return {
+            "type": "image_url",
+            "image_url": {"url": f"data:image/{self.format};base64,{self.base64_data}"},
+        }
+
+
+def get_clipboard_image() -> ImageData | None:
+    """Attempt to read an image from the system clipboard.
+
+    Supports macOS via `pngpaste` or `osascript`.
+
+    Returns:
+        ImageData if an image is found, None otherwise
+    """
+    if sys.platform == "darwin":
+        return _get_macos_clipboard_image()
+    # Linux/Windows support could be added here
+    return None
+
+
+def _get_macos_clipboard_image() -> ImageData | None:
+    """Get clipboard image on macOS using pngpaste or osascript.
+
+    First tries pngpaste (faster if installed), then falls back to osascript.
+
+    Returns:
+        ImageData if an image is found, None otherwise
+    """
+    # Try pngpaste first (fast if installed)
+    try:
+        result = subprocess.run(
+            ["pngpaste", "-"],  # Output to stdout
+            capture_output=True,
+            check=False,
+            timeout=2,
+        )
+        if result.returncode == 0 and result.stdout:
+            # Successfully got PNG data
+            try:
+                Image.open(io.BytesIO(result.stdout))  # Validate it's a real image
+                base64_data = base64.b64encode(result.stdout).decode("utf-8")
+                return ImageData(
+                    base64_data=base64_data,
+                    format="png",
```

Comment:
How do you know it's a png? (does `pngpaste` reutnrs png always?`) (maybe add a sentence to explain what it does)